pkg/backup/backup_job.go

```diff
@@ -1003,6 +896,80 @@ func (b *backupResumer) ensureClusterIDMatches(clusterID uuid.UUID) error {
 	return nil
 }
 
+func prepBackupMeta(
+	ctx context.Context,
+	execCtx sql.JobExecContext,
+	initialDetails jobspb.BackupDetails,
+	backupDest backupdest.ResolvedDestination,
+) (jobspb.BackupDetails, *backuppb.BackupManifest, error) {
+	db := execCtx.ExecCfg().InternalDB
+	var tenantSpans []roachpb.Span
+	var tenantInfos []mtinfopb.TenantInfoWithUsage
+	var err error
+	if err := db.Txn(ctx, func(ctx context.Context, txn isql.Txn) error {
+		tenantSpans, tenantInfos, err = getTenantInfo(
+			ctx, execCtx.ExecCfg().Codec, txn, initialDetails,
+		)
+		return err
+	}); err != nil {
+		return jobspb.BackupDetails{}, nil, err
+	}
+	return getBackupDetailAndManifest(
+		ctx, execCtx.ExecCfg(), tenantSpans, tenantInfos, initialDetails, execCtx.User(), backupDest,
+	)
+}
+
+// writePTSRecordOnBackup writes a protected timestamp record on the backup's
+// target spans/schema objects. It also updates the job details to store the
+// protected timestamp records UUID so that it can be persisted in the job
+// record for releasing when on completion.
+func writePTSRecordOnBackup(
```

Comment:
could this return the pts ID which the caller will bind to backup details? in general, we should avoid passing structs by reference.